state/backups/create.go

```diff
@@ -58,6 +61,23 @@ func create(args *createArgs) (_ *createResult, err error) {
 			}
 		}
 	}()
+
+	// Require an extra 5GB on top of the approximate backup size,
+	// or 20%, whatever is greater, to be free.
+	const minFree = 5 * humanize.GiByte
+	margin := uint64(float64(args.approxSpaceRequiredMib)*0.20) * humanize.MiByte
+	if margin < minFree {
+		margin = minFree
+	}
+	wantFree := uint64(args.approxSpaceRequiredMib) + margin/humanize.MiByte
+	for _, dir := range []string{builder.archivePaths.DBDumpDir, os.TempDir()} {
+		available := args.availableDisk(dir) / humanize.MiByte
+		logger.Infof("free disk on volume hosting %q: %dMiB", dir, available)
+		if available < wantFree {
```

Comment:
I think we want a --force for this, since 5gb might be prohibitive for small controllers.

Reply:
I added another check to ensure we never require more than 10% of the disk size to be free.